rfcs/convergence/authoring-stories.md

```diff
@@ -0,0 +1,290 @@
+# RFC: Authoring storybook stories
+
+---
+
+_List contributors to the proposal:_ @hotell, @miroslavstastny, @ling1726, @andrefcdias
+
+<!-- toc -->
+
+- [Summary](#summary)
+- [Problem statement](#problem-statement)
+- [Detailed Design or Proposal](#detailed-design-or-proposal)
+  - [1. what should we use for interactive props playground - controls or knobs?](#1-what-should-we-use-for-interactive-props-playground---controls-or-knobs)
+  - [2. how to use `argTypes`](#2-how-to-use-argtypes)
+  - [3. should controls work for all stories or only for general/default one](#3-should-controls-work-for-all-stories-or-only-for-generaldefault-one)
+  - [4. should we render descriptions table in Canvas Control pane](#4-should-we-render-descriptions-table-in-canvas-control-pane)
+  - [5. how to author e2e suites for those stories](#5-how-to-author-e2e-suites-for-those-stories)
+  - [6. how to properly annotate stories with TS metadata to get the best DX possible](#6-how-to-properly-annotate-stories-with-ts-metadata-to-get-the-best-dx-possible)
+  - [7. how structure stories in the storybook nav tree](#7-how-structure-stories-in-the-storybook-nav-tree)
+  - [8. dissecting big story files into smaller ones](#8-dissecting-big-story-files-into-smaller-ones)
+  - [Pros and Cons](#pros-and-cons)
+- [Discarded Solutions](#discarded-solutions)
+- [Open Issues](#open-issues)
+
+<!-- tocstop -->
+
+## Summary
+
+Currently we have no style-guide/functionality requirements on how to be consistent when writing stories for converged components. This RFC should describe (as we progress) the style we want going forward.
```

Comment:
For a little more background/context: is the intent that this storybook is the primary documentation for our controls? It might be worthwhile to mention that here, as motivation to why we want to have consistency in the storybook. And if so, it would be great to eventually make a permanent documentation authoring guide based on the decisions in this RFC.

Reply:
> is the intent that this storybook is the primary documentation for our controls?

I have _some_ historical knowledge here. The storybook is not intended to be our controls, but currently they **are** the only way we have of communicating any sort of docs with our partners AFAIK

Reply:
the intent is to have consistency and guidelines ideally for everything (not just storybook stories) for more effective collaboration and automation capabilities :) 

regarding official docs, that is still in flux, partially what @ling1726 said.  Hopefully we will leverage as much as possible from stories for final docs solution.